File: 2021/01/01.py
f = open("input.txt", "r")
l = list(map(int, f))

# Part 1
incr = -1
prev = 0
for e in l:
    if e > prev:
        incr += 1
    prev = e
print(incr)

# Part 2
incr = -1
prev = 0
for i, _ in enumerate(l):
    try:
        curr = l[i] + l[i+1] + l[i+2]
    except:
        print(incr)
        break
    if curr > prev:
        incr += 1
    prev = curr

File: 2022/01/01.rs
use std::fs::File;
use std::io::{BufRead, BufReader, Error, ErrorKind, Read};

fn read<R: Read>(io: R) -> Result<Vec<Vec<usize>>, Error> {
    let br = BufReader::new(io);
    let mut out = vec![];
    let mut curr = vec![];
    for line in br.lines() {
        let line = line?;
        if line.is_empty() {
            out.push(curr);
            curr = vec![]
        } else {
            curr.push(line.trim().parse().map_err(|e| Error::new(ErrorKind::InvalidData, e))?);
        }
    }
    Ok(out)
}

fn main() -> Result<(), Error> {
    let mut data = read(File::open("input.txt")?)?.into_iter().map(|e| e.iter().sum()).collect::<Vec<usize>>();
    data.sort();
    data.reverse();
    println!("{}", data[0]);
    println!("{}", data[..3].into_iter().sum::<usize>());
    Ok(())
}
